dpnp/backend/kernels/dpnp_krnl_arraycreation.cpp

```diff
@@ -121,35 +121,50 @@ void dpnp_ones_c(void* result, size_t size)
     dpnp_memory_free_c(fill_value);
 }
 
+template <typename _DataType>
+void dpnp_ones_like_c(void* result, size_t size)
+{
+    dpnp_ones_c<_DataType>(result, size);
+}
+
+template <typename _DataType>
+class dpnp_tri_c_kernel;
+
 template <typename _DataType>
 void dpnp_tri_c(void* result1, const size_t N, const size_t M, const int k)
 {
-    _DataType* result = reinterpret_cast<_DataType*>(result1);
+    cl::sycl::event event;
 
-    for (size_t i = 0; i < N; ++i)
-    {
-        int val = i + k + 1;
-        size_t diag_idx_ = (val > 0) ? (size_t)val : 0;
-        size_t diag_idx = (M < diag_idx_) ? M : diag_idx_;
+    _DataType* result = reinterpret_cast<_DataType*>(result1);
 
-        for (size_t j = 0; j < diag_idx; ++j)
-        {
-            size_t ind = i * M + j;
-            result[ind] = 1;
-        }
-        for (size_t j = diag_idx; j < M; ++j)
+    cl::sycl::range<1> gws(N);
+    auto kernel_parallel_for_func = [=](cl::sycl::id<1> global_id) {
+        size_t i = global_id[0];
         {
-            size_t ind = i * M + j;
-            result[ind] = 0;
+            int val = i + k + 1;
+            size_t diag_idx_ = (val > 0) ? (size_t)val : 0;
+            size_t diag_idx = (M < diag_idx_) ? M : diag_idx_;
+
+            for (size_t j = 0; j < diag_idx; ++j)
+            {
+                size_t ind = i * M + j;
+                result[ind] = 1;
```

Comment:
Just in case, is it possible here that different threads simultaneously write to the same place? I mean can we have the same `ind` for different threads?

Reply:
`ind` takes threads `id`.

Reply:
"I mean can we have the same ind for different threads?"
It is impossible by design. Each thread takes unique `id` by input parameter.